### PR TITLE
MSP Alarm support.

### PR DESCRIPTION
**Pull request by dustin**
This change creates a new MSP command that allows the peer (e.g. OSD) to ask for alarms from the flight controller.  The alarm message includes severity and a message.  In its current form, only one alarm is sent (fairly arbitrarily, other than severity taking precedence).  Also, as a special and very useful case, boot reason is considered a warning from the perspective of the MSP code for the first few seconds.  It might be worth generally considering that an alarm which would allow this special case code to be removed.

This has been very useful to me in showing both regular alarms (e.g. I let someone fly my sparky yesterday and saw 'MAN:NO RX' flashing on the screen because OpenTX companion turned off the radio for that profile) as well as boot reasons (saw watchdog firing in flight instead of mysteriously crashing).

STATUS:  Flown a lot.  Used to diagnose bugs.
CAVEAT:  MWOSD side is not yet accepted (waiting for @ShikOfTheRa in https://github.com/ShikOfTheRa/scarab-osd/pull/141)

### CODE REVIEW

File: flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c [hunk 1]
```diff
@@ -391,6 +397,154 @@ static void msp_send_boxids(struct msp_bridge *m) {
 	msp_send(m, MSP_BOXIDS, boxes, len);
 }
 
+static const char alarm_names[][8] = {
```
Comment (mluessi) [inline flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:400]: Is it necessary to define the size of the 2nd dimension?

File: flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c [hunk 2]
```diff
@@ -391,6 +397,154 @@ static void msp_send_boxids(struct msp_bridge *m) {
...
+	[SYSTEMALARMS_ALARM_ALTITUDEHOLD] = "A-HOLD",
+	[SYSTEMALARMS_ALARM_BOOTFAULT] = "BOOT",
+	[SYSTEMALARMS_ALARM_GEOFENCE] = "GEOFENCE",
+	[SYSTEMALARMS_ALARM_TEMPBARO] = "TEMPBARO",
+	[SYSTEMALARMS_ALARM_GYROBIAS] = "GYROBIAS",
+	[SYSTEMALARMS_ALARM_ADC] = "ADC",
+};
+
+// If someone adds a new alarm, we'd like it added to the array above.
+DONT_BUILD_IF(NELEMENTS(alarm_names) != SYSTEMALARMS_ALARM_NUMELEM, AlarmArrayMismatch);
```
Comment (mluessi) [inline flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:427]: Nice :)

**Comment by mluessi** [comment]
Nice work. I haven't yet had time to review the code carefully, will do so in the next few days.

File: flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c (hunk 1 above)
Comment (tracernz) [inline flight/Modules/UAVOMSPBridge/UAVOMSPBridge.c:400]: Yes, you always need to define the size of the second and subsequent dimensions.

**Pull request merged**